src/plugins/intel_gpu/src/graph/graph_optimizer/compile_graph.cpp

```diff
@@ -42,7 +42,17 @@ void compile_graph::run(program& p) {
         auto& node = *(std::next(proc_order.begin(), idx));
         const bool use_shape_agnostic_impl = !p.get_config().get_property(ov::intel_gpu::use_only_static_kernels_for_dynamic_shape);
         const impl_types original_impl_type = node->get_preferred_impl_type();
-        const bool change_initial_impl = node->is_dynamic() && original_impl_type == impl_types::onednn;
+        bool change_initial_impl = node->is_dynamic() && original_impl_type == impl_types::onednn;
+
+        if (node->is_type<fully_connected>() && change_initial_impl) {
+            const auto fc_prim = node->as<fully_connected>().get_primitive();
+            const auto weights_dt = node->get_input_layout(1).data_type;
+
+            // Do not change impl (i.e. do not use ocl shape-agnostic kernels) in case of FC and 8bit compressed weights,
+            // since oneDNN primitives/kernels caching mechanism will be used instead.
+            if (fc_prim->compressed_weights && ov::element::Type(weights_dt).bitwidth() == 8)
+                change_initial_impl = false;
```

Comment:
Perf check for this change done? If there is, could you share the record?

Reply:
@yeonbok we have asked validation team to check these changes, included you to the copy of message